### DISCUSSION TITLE
Playlist Creation Improvements + Layout Overhaul

### DISCUSSION THREAD

**Discussion by WalmartWhiteCat** [ppy/osu#27072] [Ideas]
**Preamble:** I think we can all agree that playlists are _pretty damn cool_. BUT..! They could be way better. The changes I’m mainly thinking of would probably be a Supporter benefit, as they kind of sit in that premium range of osu! stuff, but they don’t have to be.

- Allow supporters to set a time and date down to the exact second that the playlist will be created. For non-supporters, allow them to create a playlist at their current time only.

- Allow supporters to have the playlist be renewed (Everything resets) on intervals. An example would be allowing them to set it so the playlist is renewed at the start of every month/2 weeks/week/day. Depending on how often the playlist is renewed, limit the creator down to that for how long it can last (i.e. a playlist that comes out at the start of every week can only be a week long. If it comes out every 2 weeks then it can only be 2 weeks long, etc…). Also make it so the creator can choose how many times the playlist is renewed.

- Allow supporters to add a reward to the playlist and choose who that reward goes to, as well as the interval at which the reward goes out (i.e. first place and second place get Supporter on Friday at 5pm PST every two weeks). Supporter doesn’t have to be the only reward, there could also be banners or profile pictures that would be automatically messaged to the winner. There could also be the chance for a profile badge, but maybe make it so that the playlist needs to be verified by a Moderator first or something.

- When a playlist is set to renew on intervals, make it so the info and maps can still be changed for the next renewal, or that defeats the whole point of automation. Alternatively, make it so creators can set up a queue of renewals (i.e. Sourced’s Awesome Playlist with 5 maps, followed by Hitsounder’s Awesome Playlist with 10 maps and a Supporter reward). If a queue is not set up and nothing is changed, then it will simply make the exact same playlist at the next interval with a wiped leaderboard.

- Allow all creators to set an owner message meant to explain a bit of what the playlist will be like, what it’s for, how it works and stuff like that. Make it so supporters can add multiple messages (Probably up to 3 or something) that rotate every 5-10 seconds or upon being clicked. Also allow these messages to be changed during the runtime of the playlist (They don’t change on intervals).

I think it would also be a good idea to overhaul the look of playlists, as the layout right now is fine, but it seems kinda… meh. I already made a sort of prototype, but the design does not have to be used.

![osu!LazerPlaylistAlternativeNoDev](https://github.com/ppy/osu/assets/149839359/d2e057eb-a759-451b-a208-73a87f76ffd5)
(Not dev notes)
![osu!LazerPlaylistAlternative](https://github.com/ppy/osu/assets/149839359/e1f0024e-ab7a-474d-9404-147b3f1f6c8f)
(Dev Notes)

**Closing thoughts:** I don't make this post to say that playlists are bad or anything, I absolutely love them, they're super fun. I'm just an enthusiast who likes to throw ideas around when I can :)

**Comment by Stoppedpuma** [comment]
Designs are already being made: https://www.figma.com/file/XJCRjfvTlVIwZBWoAz5kB2/Client%2FPlaylist-Multiplayer?type=design&node-id=0-1&mode=design&t=8KFzTqdIOrs7b9cH-0.

**Comment by Rekunan** [comment]
If any were to become supporter only, only the 2nd one makes sense, and maybe the 3rd for non-supporter only rewards. The 1st doesn't change how much u can create, and the 5th is just a small amount of text.
